Connect programs report table to API data

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Separate category-agnostic `CompareTableCard` (eda8cc4, 01df46e, bf32393)
- Use it in `CompareProductsTableCard` (bf32393)
- Use it in `CompareProgramsTableCard` and connect it to API data (0dde6b3)

Other changes:
- Unify `trackEventId` property name between programs and products report pages (2af2f06)
- Unify `isLoading` parametter across all `<*TableCard>`s. (bf32393)
- Add support for quering programs report by ids. (7e7af54)

Implements part of https://github.com/woocommerce/google-listings-and-ads/issues/242

### Screenshots:

![Programs report table](https://user-images.githubusercontent.com/17435/118701853-7a8f4b00-b814-11eb-96b6-a4b8266348f9.gif)



### Detailed test instructions:
0. Make sure you have some test data, or use the mock:
```php
add_filter(
	'gla_prepared_response_ads-reports-products',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/ads/reports/products.json' ), true ) ?: [];
	},
);
add_filter(
	'gla_prepared_response_mc-reports-products',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/mc/reports/products.json' ), true ) ?: [];
	},
);
add_filter(
	'gla_prepared_response_ads-reports-programs',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/ads/reports/programs.json' ), true ) ?: [];
	},
);
add_filter(
	'gla_prepared_response_mc-reports-programs',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/mc/reports/programs.json' ), true ) ?: [];
	},
);
```
1. Go to https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms
2. Check the data in the table
3. Filter programs "All Google Programs", "Single: Free", "Single: Campa", "Compare"
4. Check that unavailable columns collapse when you query only for free listings.
4. Check that "Unavailable" is shown, if programs are missing some data if you query for many programs.
5. Check that once you select programs to compare respective `ids` are being sent in `/wp-json/wc/gla/ads/reports/programs` requests

:warning:

1.  Make sure the Product table works as before.

### Changelog Note:

> Connected programs report table to API data.

### :scroll: To be done here, or in a separate issue:
1. Make report hooks adapt the request and returned data when the user selects to order by a field that's not available for a free listing.

tomalec committed:
Restore original `CompareProductsTableCard`

tomalec committed:
Unify `trackEventId` property name

between programs and products report pages.

tomalec committed:
Move `CompareProductsTableCard` to shared `.~/reports/` folder

tomalec committed:
Make `CompareTableCard` report category agnostic,

Make `CompareProductsTableCard` use `CompareTableCard`.

Unify `isLoading` parametter across all `<*TableCard>`s.
Make cell rendered display "Unavailable" if any data is missing - prepare to be reused in Programs report.

tomalec committed:
Connect `CompareProgramsTableCard` to API data,

use  specialized `CompareTableCard`

tomalec committed:
Add support for quering programs report by ids.

eason9487 reviewed (approved):
Tested locally and works well. The extraction of the shared **CompareTableCard** is looking good! 💯 

Left some comments, mostly about typos and some minor doubts, and not a blocker for this PR.

tomalec committed:
Remove redundant `totals` check.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/630/files#r635068894 and https://github.com/woocommerce/google-listings-and-ads/pull/630/files#r635103609

tomalec committed:
Fix `ProgramsData.name` JSDoc description.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/630/files#r635063382

tomalec committed:
Fix product/program/report naming in JSDoc comments

Apply suggestions from code review

Co-authored-by: Eason <eason.su.tw@gmail.com>
